src/main/java/de/muenchen/isi/domain/service/AbfrageService.java

```diff
@@ -65,7 +65,12 @@ public InfrastrukturabfrageModel saveInfrastrukturabfrage(final Infrastrukturabf
         var abfrageEntity = this.abfrageDomainMapper.model2entity(abfrage);
         var saved = this.infrastrukturabfrageRepository.findByAbfrage_NameAbfrageIgnoreCase(abfrageEntity.getAbfrage().getNameAbfrage());
         if(saved.isPresent()) {
-            throw new UniqueViolationException("Der angegebene Name der Abfrage ist schon vorhanden, bitte wählen Sie daher einen anderen Namen und speichern Sie die Abfrage erneut.");
+            if (abfrageEntity.getId().equals(saved.get().getId())) {
+                abfrageEntity = this.infrastrukturabfrageRepository.save(abfrageEntity);
```

Comment:
Die verschachtelte If-Anweisung ist etwas unübersichtlich. Des Weiteren wird die selbe Speicheroperation und das Mapping an zwei stellen durchgeführt.

```java
if ((saved.isPresent() && bauvorhabenEntity.getId().equals(saved.get().getId())) || !saved.isPresent()) {

            bauvorhabenEntity = this.bauvorhabenRepository.save(bauvorhabenEntity);
            return this.bauvorhabenDomainMapper.entity2Model(bauvorhabenEntity);
} else {
          throw new ....
}
```

Reply:
Da es aktuell nur in den zwei Fällen vorkommt, halte ich es nicht für nötig für ein if-Statement eine komplette Methode auszulagern, da man bei dieser dann generisch eh die Typen mit übergeben müsste.